### ISSUE TITLE
need to change the importance algorithm

### ISSUE THREAD

**Issue by marcelsalathe** [crowdAI/crowdai#629]
Our current home page shows three challenges that have ended, while there are multiple ones that are currently running. This makes no sense and it's difficult to market crowdAI when all we show are past challenges. What is the current algorithm?

**Comment by scarroll32** [comment]
## TL; DR

I suggest we modify sort orders as follows:
- featured sequence is always the highest sort order, regardless of status
- When a challenge ends, it's featured sequence is removed automatically
- For ended challenges we can override and add the featured sequence back in (but then have to remove it later)
- Change to use challenge start date as tertiary sort field, instead of current challenge creation date (does not apply to featured sequenced challenges)


## Longer version

<img width="474" alt="screen shot 2018-03-22 at 09 25 20" src="https://user-images.githubusercontent.com/11340230/37760817-13bffb52-2db8-11e8-9ead-92f2476d8ebc.png">


We actually have 3 different sort orders, for the: 

- landing page (sorts by featured sequence, descending)
- challenges (All) page (by status then 
- challenges (Active) page

Without featured sequence, order is: **running, starting soon, completed**



Landing Page:

- Keep the **featured sequence** as the highest sort order (higher is higher on the page). This is fine grained control over the challenge's position if it is running or not.
- If there is not any or less than 5 entries with **featured sequence**, then sort by challenge start date descending)
- If a challenge becomes completed, then the featured sequence is removed automatically. We can then manually add the **featured sequence** back in if we want to pin a completed or not started challenge on the landing page

All Challenges view

- Sort by **status**, then by **challenge start date**. Currently we use creation date, but this is problematic as we sometimes draft challenges a long time before they start.
- For most users they will see only **running, starting soon, completed** in that order
- If a challenge has the featured sequence on it, it overrides the status (goes to the top of the list)

Active Challenges view

- Same as All Challenges, but only includes **running** and **starting soon**

  CASE status_cd
              WHEN 'running' THEN 1
              WHEN 'starting_soon' THEN 2
              WHEN 'completed' THEN 3
              WHEN 'cancelled' THEN 4
              WHEN 'draft' THEN 5
              ELSE 6
            END, created_at DESC"

**Comment by spMohanty** [comment]
I would also reference this issue : https://github.com/crowdAI/crowdai/issues/576

Where we wanted to use “participation” as one of the criterias for sorting ! We would want to bubble up all the challenges with a high participation.

**Comment by scarroll32** [comment]
@spMohanty OK, good idea.

For non-featured sequence challenges, we could sort by # of participants, descending, instead of start date ?

Featured sequence will still be a global override.

**Comment by spMohanty** [comment]
Sounds good! But challenge status still gets a priority nevertheless, so running challenges get a priority.

**Comment by scarroll32** [comment]
So the final sort order would be:

1. Challenges with a **featured sequence**, sorted by featured sequence descending
2.  Challenges without a featured sequence, by **running, starting soon, completed** status in that order
3. within **running, starting soon, completed**, sort by number of participants descending

- We won't use the challenge creation or start date at all 
- When a challenge automatically moved from running -> completed, the **featured sequence** is cleared. It can be re-added manually and then must be re-cleared manually.

@spMohanty @marcelsalathe agreed?

**Comment by marcelsalathe** [comment]
Agreed.

**Comment by scarroll32** [comment]
Implemented
